File: 0206-reverse-linked-list/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* tmp=nullptr;
        if(head==nullptr)return head;
        ListNode* curr=head;
        ListNode* nxt=head->next;
        while(1){
            curr->next=tmp;
            if(nxt==nullptr)return curr;
            tmp=curr;
            curr=nxt;
            nxt=nxt->next;
        }
    }
};

File: 0206-reverse-linked-list/solution.cs
public class Solution {
    public ListNode ReverseList(ListNode head) {
        if(head == null) return head;
        ListNode curr = head;
        ListNode pre = null;
        ListNode nxt = curr.next;
        while(nxt != null){
            curr.next = pre;
            pre = curr;
            curr = nxt;
            nxt = nxt.next;
        }
        curr.next = pre;
        return curr;
    }
}
